json_serializable/lib/src/utils.dart

```diff
@@ -11,6 +11,42 @@ import 'package:analyzer/src/dart/resolver/inheritance_manager.dart'
 
 import 'package:source_gen/source_gen.dart';
 
+// TODO: still need handle tripe singe/double quotes within `value`
+String escapeDartString(String value) {
+  if (value.contains('\n')) {
+    return "r'''\n$value'''";
+  }
+
+  var containsSingleQuote = value.contains("'");
+  var contains$ = value.contains(r'$');
+
+  if (containsSingleQuote) {
+    if (value.contains('"')) {
+      // `value` contains both single and double quotes as well as `$`.
+      // The only safe way to wrap the content is to escape all of the
+      // problematic characters.
+      var string = value
+          .replaceAll('\$', '\\\$')
```

Comment:
Inconsistent between `'\$'` and `r'$'` from line 21. How about `.replaceAll(r'$', r'\$')`

Similarly for the quotes I think it's more readable with `r` than with double slash.